sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.aest.controller.updateProfileInformation.e2e-spec.ts

```diff
@@ -0,0 +1,181 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { IdentityProviders } from "@sims/sims-db";
+
+describe("StudentAESTController(e2e)-getUploadedFile", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should allow the student profile update when the student is a Basic BCeID user.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    student.user.identityProviderType = IdentityProviders.BCeIDBasic;
+    await db.student.save(student);
+    const studentProfileUpdateInfo = {
+      givenNames: "SCOOBY",
+      lastName: "DOOBY-DOO",
+      birthdate: "1980-03-24",
+      email: "dummy@some.domain",
+      noteDescription: "Some dummy note.",
+    };
+    const token = await getAESTToken(AESTGroups.OperationsAdministrators);
+    let endpoint = `/aest/student/${student.id}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(studentProfileUpdateInfo)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+
+    // Arrange
+    endpoint = `/aest/student/${student.id}`;
+    // Act/Assert
+    const response = await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+    expect(response.body).toEqual(
+      expect.objectContaining({
+        firstName: "SCOOBY",
+        lastName: "DOOBY-DOO",
+        dateOfBirth: "1980-03-24",
+        email: "dummy@some.domain",
+      }),
+    );
+  });
+
+  it("Should throw a HTTP Unprocessable Entity (422) error when the student is not a Basic BCeID user.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    student.user.identityProviderType = IdentityProviders.BCeIDBusiness;
+    await db.student.save(student);
+    const studentProfileUpdateInfo = {
+      givenNames: "SCOOBY",
+      lastName: "DOOBY-DOO",
+      birthdate: "1980-03-24",
+      email: "dummy@some.domain",
+      noteDescription: "Some dummy note.",
+    };
+    const token = await getAESTToken(AESTGroups.OperationsAdministrators);
+    const endpoint = `/aest/student/${student.id}`;
+
+    // Act/Assert
+    const response = await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(studentProfileUpdateInfo)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
Please follow the below as it is the most used along the assertions and it seems cleaner. Same for other tests if required.
```ts
.expect(HttpStatus.UNPROCESSABLE_ENTITY)
.expect({
      error: "Unprocessable Entity",
      message: "Not possible to update a non-basic-BCeID student.",
      statusCode: HttpStatus.UNPROCESSABLE_ENTITY,
    });
```